File: src/main/java/net/unaussprechlich/managedgui/lib/util/RenderHelper.java
/*
 * ***************************************************************************
 *
 *         Copyright © 2016 unaussprechlich - ALL RIGHTS RESERVED
 *
 * ***************************************************************************
 */
package net.unaussprechlich.managedgui.lib.util;

import net.minecraft.util.ResourceLocation;
import net.unaussprechlich.managedgui.lib.elements.Container;

public class RenderHelper {

    public static void renderContainer(Container container) {

        renderContainerBackground(
                container.getXStartPadding(),
                container.getYStartPadding(),
                container.getBackgroundRGBA(),
                container.getWidthPadding(),
                container.getHeightPadding(),
                container.getBackgroundImage()
        );

        RenderUtils.renderBorder(
                container.getXStartBorder(),
                container.getYStartBorder(),
                container.getWidthBorder(),
                container.getHeightBorder(),
                container.getBorder(),
                container.getBorderRGBA()
        );
    }

    private static void renderContainerBackground(int xStart, int yStart, ColorRGBA color, int width, int height, ResourceLocation image) {
        if (image == null) RenderUtils.renderBoxWithColor(xStart, yStart, width, height, color);
        else RenderUtils.drawModalRectWithCustomSizedTexture(xStart, yStart, width, height, image, color.getALPHAf());
    }

}

File: src/main/java/net/unaussprechlich/managedgui/lib/util/RenderHelper.kt
/*
 * ***************************************************************************
 *
 *         Copyright © 2016 unaussprechlich - ALL RIGHTS RESERVED
 *
 * ***************************************************************************
 */
package net.unaussprechlich.managedgui.lib.util

import net.minecraft.util.ResourceLocation
import net.unaussprechlich.managedgui.lib.container.Container

object RenderHelper {

    fun renderContainer(container: Container) {

        renderContainerBackground(
                container.xStartPadding,
                container.yStartPadding,
                container.backgroundRGBA,
                container.widthPadding,
                container.heightPadding,
                container.backgroundImage
        )

        if(!container.border.isEmpty)
        RenderUtils.renderBorder(
                container.xStartBorder,
                container.yStartBorder,
                container.widthBorder,
                container.heightBorder,
                container.border,
                container.borderRGBA
        )
    }

    private fun renderContainerBackground(xStart: Int, yStart: Int, color: ColorRGBA, width: Int, height: Int, image: ResourceLocation?) {
        if (image == null) RenderUtils.renderBoxWithColor(xStart, yStart, width, height, color)
        else RenderUtils.texture_modularRect(xStart , yStart, width, height, image)
    }

}
